### PR TITLE
Changes to send omsagent/omsagent-rs kubectl logs to App Insights

### PR DESCRIPTION
**Pull request by r-dilip**

### CODE REVIEW

File: installer/conf/td-agent-bit.conf [hunk 1]
```diff
@@ -14,6 +14,15 @@
     Path_Key filepath
     Skip_Long_Lines On
 
+[INPUT]
+    Name tail
+    Tag oms.container.log.flbplugin.*
+    Path /var/log/containers/omsagent*.log
+    DB /var/opt/microsoft/docker-cimprov/state/fblogs.db
```
Comment (vishiy) [inline installer/conf/td-agent-bit.conf]: do you know if this is best practice (specifying & using and same state db  for 2 input tails?). we need to make sure this telemetry enablement does not impact log collection. i would rather use a different db .

File: installer/conf/td-agent-bit.conf [hunk 2]
```diff
@@ -14,6 +14,15 @@
     Path_Key filepath
     Skip_Long_Lines On
 
+[INPUT]
+    Name tail
+    Tag oms.container.log.flbplugin.*
```
Comment (vishiy) [inline installer/conf/td-agent-bit.conf]: the oms.container.log.* tails omsagent*.log as well. its that we need to send this to our telemetry AI (and also to customers workspace if kube-sys log collection is enabled). should we need anither tailer which is doing subset of the already exsting one? In this case, we can simply check the file name and route to AI (and also to workspace if kube-sys collection is enabled).

File: source/code/go/src/plugins/out_oms.go [hunk 3]
```diff
@@ -50,6 +50,12 @@ func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
 		}
 		records = append(records, record)
 	}
+
+	incomingTag := C.GoString(tag)
+	if strings.Contains(strings.ToLower(incomingTag), "flbplugin") {
```
Comment (vishiy) [inline source/code/go/src/plugins/out_oms.go]: pls check for full tag

File: source/code/go/src/plugins/telemetry.go [hunk 4]
```diff
@@ -103,51 +146,14 @@ func initialize(telemetryPushIntervalProperty string, agentVersion string) (int,
...
-	TelemetryClient.Track(event)
-}
-
-// SendException  send an event to the configured app insights instance
-func SendException(err interface{}) {
-	if TelemetryClient != nil {
-		TelemetryClient.TrackException(err)
-	}
+	traceEntry := strings.Join(logLines, "\n")
+	TelemetryClient.TrackTrace(traceEntry, 1)
```
Comment (vishiy) [inline source/code/go/src/plugins/telemetry.go]: do u know if there is a size limit imposed by AI for traceEntry in TrackTrace?

File: installer/conf/td-agent-bit.conf (hunk 1 above)
Comment (r-dilip) [reply installer/conf/td-agent-bit.conf]: if you look at the db, it is basically a set of files, along with the last position that was read. So, in my opinion, should be ok. that being said, will do more testing to see if there is any impact to log collection

File: installer/conf/td-agent-bit.conf (hunk 2 above)
Comment (r-dilip) [reply installer/conf/td-agent-bit.conf]: As discussed offline, this way, we can keep the logs and telemetry separate by having two separate tailers

File: source/code/go/src/plugins/out_oms.go (hunk 3 above)
Comment (r-dilip) [reply source/code/go/src/plugins/out_oms.go]: Will do.

**Pull request merged**